mlrun/features.py

```diff
@@ -410,9 +418,12 @@ def __init__(self, check_type: bool = None, severity: str = None, regex=None):
         """
         super().__init__(check_type, severity)
         self.regex = regex
-        self.regex_compile = re.compile(self.regex)
+        self.regex_compile = None
```

Comment:
I think you can set it here if `self.regex` is not None and inherit `from_dict` and create it there in case it is None. That way you won't need to ask if every time in the `check` method.